src/attr.rs

```diff
@@ -171,6 +180,31 @@ impl<'s> Attributes<'s> {
         }
     }
 
+    fn insert_str(&mut self, key: &'s str, val: &'s str) -> usize {
+        if self.0.is_none() {
+            self.0 = Some(Vec::new().into());
+        };
```

Comment:
this function is identical to `insert` above except that it takes a str instead of attributevalue. possible to share code here?